Expose CXX_FLAGS through __config__

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47865 Enable callgrind collection for C++ snippets
* #47864 Add support for timing C++ snippets.
* #47863 Rework compat bindings.
* #47862 Add option for cpp_extensions to compile standalone executable
* **#47861 Expose CXX_FLAGS through __config__**

Differential Revision: [D25199263](https://our.internmc.facebook.com/intern/diff/D25199263)

Commit:
Expose CXX_FLAGS through __config__

[ghstack-poisoned]

robieta commented:
@pytorchbot retest this please

Commit:
Update on "Expose CXX_FLAGS through __config__"




[ghstack-poisoned]

Commit:
Update on "Expose CXX_FLAGS through __config__"




[ghstack-poisoned]

Commit:
Update on "Expose CXX_FLAGS through __config__"




[ghstack-poisoned]

Commit:
Update on "Expose CXX_FLAGS through __config__"


Differential Revision: [D25199263](https://our.internmc.facebook.com/intern/diff/D25199263)

[ghstack-poisoned]

Commit:
Update on "Expose CXX_FLAGS through __config__"


Differential Revision: [D25199263](https://our.internmc.facebook.com/intern/diff/D25199263)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/Version.cpp:
```diff
@@ -186,6 +186,9 @@ std::string show_config() {
 
 std::string get_cxx_flags() {
+  #if defined(FBCODE_CAFFE2)
+  TORCH_CHECK(false, "CXX_FLAGS is only for OSS build.");
```
This error message is not as descriptive as it could be. Why is CXX_FLAGS only for OSS build?

robieta replied:
```Buck does not populate the `CXX_FLAGS` field of Caffe2 build options. As a result, get_cxx_flags is OSS only.```?

Commit:
Update on "Expose CXX_FLAGS through __config__"


Differential Revision: [D25199263](https://our.internmc.facebook.com/intern/diff/D25199263)

[ghstack-poisoned]
